sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts

```diff
@@ -166,144 +170,242 @@ describe("EducationProgramOfferingAESTController(e2e)-assessOfferingChangeReques
+    expect(queryRequestedOffering.offeringNote.description).toBe(
+      "offering change approved",
+    );
+    const queryApplications = await db.application.find({
+      select: {
+        id: true,
+        currentAssessment: {
+          triggerType: true,
+          offering: { id: true },
         },
-        where: { id: precedingOffering.id },
-      });
-      const queryRequestedOffering = await db.educationProgramOffering.findOne({
-        select: {
-          id: true,
-          offeringStatus: true,
-          offeringNote: { id: true, description: true },
+      },
+      relations: {
+        currentAssessment: { offering: true },
+      },
+      where: {
+        currentAssessment: {
+          offering: { id: requestedOffering.id },
+        },
+      },
+    });
+    expect(queryApplications.length).toBe(0);
+  });
+
+  it("Should determine the offering change for an offering with one or more applications in Completed status when the offering change is approved.", async () => {
+    // Arrange
+    for (let i = 0; i < 2; i++) {
+      const application = await saveFakeApplication(
+        db.dataSource,
+        {
+          institution: collegeFLocation.institution,
+          institutionLocation: collegeFLocation,
         },
-        relations: { offeringNote: true },
-        where: { id: requestedOffering.id },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+        },
+      );
+      // Create a student appeal for the application and its student assessment.
+      const studentAppeal = createFakeStudentAppeal({
+        application: application,
+        studentAssessment: application.currentAssessment,
       });
-      expect(queryPrecedingOffering.offeringStatus).toBe(
-        OfferingStatus.ChangeOverwritten,
+      application.currentAssessment.studentAppeal = studentAppeal;
+      application.currentAssessment.offering = precedingOffering;
+      await db.application.save(application);
+    }
+
+    const payload: OfferingChangeAssessmentAPIInDTO = {
+      offeringStatus: OfferingStatus.Approved,
+      assessmentNotes: "offering change approved",
+    };
+    // Ministry token.
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    const endpoint = `/aest/education-program-offering/${requestedOffering.id}/assess-change-request`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+
+    const queryApplications = await db.application.find({
+      select: {
+        id: true,
+        currentAssessment: {
+          triggerType: true,
+          offering: { id: true },
+          studentAppeal: { id: true },
+        },
+      },
+      relations: {
+        currentAssessment: { offering: true, studentAppeal: true },
+      },
+      where: {
+        currentAssessment: {
+          offering: { id: requestedOffering.id },
```

Comment:
Even though it is existing logic, I would recommend to get the applications based in applicationIds like the recent test. Other wise this test will not be independent by itself.